src/main/java/seedu/duke/controllers/ModulePlannerController.java

```diff
@@ -0,0 +1,248 @@
+
+
+    }
+
+
+    /**
+     * Starts the interactive command-line interface for the academic module management system.
+     * This method displays a welcome message, reads user input, and processes various commands.
+     * While the user input is not "bye," the method processes the input and responds accordingly.
+     * The commands are case-insensitive, and the response is displayed in the view.
+     */
+    public void start() {
+        view.displayWelcome();
+        Scanner in = new Scanner(System.in);
+        String userInput = in.nextLine();
+
+        while (!userInput.equals("bye")) {
+
+            String[] words = userInput.split(" ");
+
+            String initialWord = words[0].toLowerCase();
+
+            switch (initialWord) {
+            case "hi": {
+                view.displayMessage("can put the commands here");
+                break;
+            }
+            case "hello": {
+                view.displayMessage("yup");
+                break;
+            }
+            case "info": {
+                view.displayMessage("info");
+                JSONObject moduleInfoObject = Api.getFullModuleInfo("CS2113");
+                assert(moduleInfoObject != null);
+                String moduleInfo = (String) moduleInfoObject.get("description");
+                view.displayMessage(moduleInfo);
+                break;
+            }
+            case "left": {
+                ArrayList<String> modules = listModulesLeft();
+
+                view.displayMessage("Modules left:");
+                for (String module : modules) {
+                    view.displayMessage(module);
+                }
+                break;
+            }
+            case "pace": {
+                //assumed that everyone graduates at y4s2
+                //waiting for retrieving logic
+                int modulesCreditsCompleted = 100;
+                int totalCreditsToGraduate = 160;
+                int creditsLeft = totalCreditsToGraduate - modulesCreditsCompleted;
+                computePace(words, creditsLeft);
+                break;
+            }
+            case "prereq": {
+                if(!Parser.isValidInput("prereq",words)){
+                    ErrorHandler.invalidInput();
+                    break;
+                }
+                String keyword = words[1];
+                System.out.println(getModulePrereqBasedOnCourse(keyword.toUpperCase(), "CEG"));
+                break;
+            }
+            case "major": {
+                String printMessageCommand = student.updateMajor(userInput);
+                switch (printMessageCommand) {
+                case "currentMajor":
+                    view.displayMessage("Current major is " + student.getMajor() + ".");
+                    break;
+                case "newMajor":
+                    view.displayMessage("Major " + student.getMajor() + " selected!");
+                    break;
+                case "invalidMajor":
```

Comment:
Consider abstracting out into different Command classes. Extensive use of magic literals here, try to avoid them.